python/pyspark/sql/connect/types.py

```diff
@@ -139,7 +139,7 @@ def pyspark_types_to_proto_types(data_type: DataType) -> pb2.DataType:
     if isinstance(data_type, NullType):
         ret.null.CopyFrom(pb2.DataType.NULL())
     elif isinstance(data_type, StringType):
-        ret.string.CopyFrom(pb2.DataType.String())
+        ret.string.collation_id = data_type.collationId
```

Comment:
Is this change tested? The added UT seems to test proto -> type conversion but not vice verse?

Reply:
@amaliujia New changes should cover conversions in both directions (both get invoked during testing), please check.